src/common/http/post_lambda_request.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#include "httpc.hpp"
+#include <functional>
+
+namespace http {
+/**
+* @brief Post Request with 'write_body_chunk' given by user via a lambda
+*
+*/
+class PostLambda : public Request {
+public:
+    using LambdaT = std::function<std::variant<size_t, Error>(char *, size_t)>;
```

Comment:
Is that thing using dynamic allocation internally? If so, is it better to have the class as a template?

Reply:
I want to allow the user to pass in a stateful lambda, so it needs  to be std::function, and std::function cannot be a template parameter

Reply:
Huh? Why couldn't you use that stateful lambda without first wrapping it into a `std::function`?

Ah, I see. It could be a template parameter, but one can't _name_ the parameter. So, you couldn't do:

```cpp
PostLambda<???> post([&](){ ... });
```

Because that `???` type can't be named. So you'd need a helper function and would have to do something like

```cpp
auto post(make_post([&](){ ... });
```

Which starts getting a bit annoying. I do believe you can have `std::function<whatever>` as a template parameter, but then, there's no actual point, because you'd still have the `std::function`.